cmd/kube-proxy/app/options.go

```diff
@@ -328,6 +315,50 @@ func (o *Options) processV1Alpha1Flags(fs *pflag.FlagSet) {
 	if fs.Changed("cluster-cidr") {
 		o.config.DetectLocal.ClusterCIDRs = strings.Split(o.clusterCIDRs, ",")
 	}
+	if fs.Changed("healthz-bind-address") {
+		host, port, err := net.SplitHostPort(o.healthzBindAddress)
+		var ip net.IP
+
+		// o.healthzBindAddress can have both "ip" and "ip:port" as value,
+		// we need to handle both cases.
+		if err != nil {
+			ip = netutils.ParseIPSloppy(o.healthzBindAddress)
```

Comment:
what if this is an invalid ip? the ip will be nil, I think you need to assert on this and fail if that is the case

Reply:
We use `utilflag.IPPortVar` to load the value of this flag and ``utilflag.IPPortVar` handle the case you mentioned.